components/formats-api/src/loci/formats/FormatTools.java

```diff
@@ -1387,53 +1387,166 @@ public static void convert(IFormatReader input, IFormatWriter output,
-      if (unit != null) {
-        try {
-          UnitsLength ul = UnitsLength.fromString(unit);
-          return UnitsLength.create(value, ul);
-        } catch (EnumerationException e) {
-        }
-      }
-      return new Length(value, UNITS.MICROM);
+      return getPhysicalSize(value, unit);
     }
     LOGGER.debug("Expected positive value for PhysicalSizeX; got {}", value);
     return null;
   }
-  
+
+  /**
+   * Formats the input value for the physical size in Y into a length of the
+   * given unit.
+   *
+   * @param value  the value of the physical size in X
+   * @param unit   the unit of the physical size in X
+   *
+   * @return       the physical size formatted as a {@link Length}
+   */
+  public static Length getPhysicalSizeX(Double value, ome.units.unit.Unit<ome.units.quantity.Length> unit) {
+    if (value != null && value - Constants.EPSILON > 0 &&
+      value < Double.POSITIVE_INFINITY)
+    {
+      createLength(value, unit);
+    }
+    LOGGER.debug("Expected positive value for PhysicalSizeX; got {}", value);
+    return null;
+  }
+
+  /**
+   * Formats the input value for the physical size in Y into a length in
+   * microns
+   *
+   * @param value  the value of the physical size in Y in microns
+   *
+   * @return       the physical size formatted as a {@link Length}
+   */
   public static Length getPhysicalSizeY(Double value) {
-    return getPhysicalSizeY(value, null);
+    return getPhysicalSizeY(value, UNITS.MICROM);
   }
-  
+
+  /**
+   * Formats the input value for the physical size in Y into a length of the
+   * given unit.
+   *
+   * @param value  the value of the physical size in Y
+   * @param unit   the unit of the physical size in Y. If {@code null},
+   *               default to microns.
+   *
+   * @return       the physical size formatted as a {@link Length}
+   */
   public static Length getPhysicalSizeY(Double value, String unit) {
     if (value != null && value - Constants.EPSILON > 0 &&
       value < Double.POSITIVE_INFINITY)
     {
-      if (unit != null) {
-        try {
-          UnitsLength ul = UnitsLength.fromString(unit);
-          return UnitsLength.create(value, ul);
-        } catch (EnumerationException e) {
-        }
-      }
-      return new Length(value, UNITS.MICROM);
+      return getPhysicalSize(value, unit);
     }
     LOGGER.debug("Expected positive value for PhysicalSizeY; got {}", value);
     return null;
   }
 
+  /**
+   * Formats the input value for the physical size in Y into a length of the
+   * given unit.
+   *
+   * @param value  the value of the physical size in Y
+   * @param unit   the unit of the physical size in Y
+   *
+   * @return       the physical size formatted as a {@link Length}
+   */
+  public static Length getPhysicalSizeY(Double value, ome.units.unit.Unit<ome.units.quantity.Length> unit) {
+    if (value != null && value - Constants.EPSILON > 0 &&
+      value < Double.POSITIVE_INFINITY)
+    {
+      createLength(value, unit);
```

Comment:
This line is missing a `return`! Maybe time to add unit tests for these methods?